extensions/blocks/instagram-gallery/edit.js

```diff
@@ -58,22 +62,37 @@ const InstagramGalleryEdit = props => {
 		apiFetch( {
 			path: addQueryArgs( '/wpcom/v2/instagram/gallery', {
 				access_token: accessToken,
-				count,
+				count: MAX_IMAGE_COUNT,
 			} ),
-		} ).then( response => {
+		} ).then( ( { external_name: externalName, images: imageList } ) => {
 			setIsLoadingGallery( false );
 
-			if ( isEmpty( response.images ) ) {
+			if ( isEmpty( imageList ) ) {
 				noticeOperations.createErrorNotice(
 					__( 'No images were found in your Instagram account.', 'jetpack' )
 				);
 				return;
 			}
 
-			setAttributes( { instagramUser: response.external_name } );
-			setImages( response.images );
+			setAttributes( { instagramUser: externalName } );
+			setImages( imageList );
 		} );
-	}, [ accessToken, count, noticeOperations, setAttributes ] );
+	}, [ accessToken, noticeOperations, setAttributes ] );
+
+	useEffect( () => {
+		noticeOperations.removeAllNotices();
+
+		if ( images.length < count ) {
```

Comment:
Maybe we should also check if the block is connected here, to avoid displaying a notice that you have 0 posts in your account before you've connected it.

Reply:
Done - haven't used _n before so hopefully I use it correctly 😃